src/gui/widgets/settings.cpp

```diff
@@ -570,15 +577,13 @@ state_definition::state_definition(const config& cfg) : canvas_()
 	canvas_.set_cfg(draw);
 }
 
-void register_widget(const std::string& id,
-					 std::function<void(gui_definition& gui,
-										  const std::string& definition_type,
-										  const config& cfg,
-										  const char* key)> functor)
+void register_widget(const std::string& id, std::function<styled_widget_definition_ptr(const config&)> f, const char* key)
 {
-	registred_widget_type().insert(std::make_pair(id, functor));
+	registred_widget_type()[id] = {f, key};
```

Comment:
Note that this changes the semantics from "first takes priority" to "last takes priority" - is that really wanted?

(Also, please remove those three useless lines below.)